File: android/wear/src/main/java/com/lowerbackstretching/wear/model/WatchModels.kt
package com.lowerbackstretching.wear.model

import com.lowerbackstretching.core.player.Timed
import kotlinx.serialization.Serializable

/**
 * Slim mirror of the phone-side [com.lowerbackstretching.core.model.Stretch].
 * The watch only needs the id, name, and duration to drive the timer;
 * the video, description, educational cards, and so on stay phone-only.
 *
 * Implements [Timed] so the shared [com.lowerbackstretching.core.player.PlayerEngine]
 * can drive either kind of routine.
 */
@Serializable
data class WatchStretch(
    val id: String,
    val name: String,
    override val durationSeconds: Int,
) : Timed

@Serializable
data class WatchRoutine(
    val name: String,
    val stretches: List<WatchStretch>,
)

File: ios/WatchApp/Models/WatchModels.swift
import Foundation

/// Slim mirror of the iPhone-side `Stretch`. The watch only needs the
/// id, name, and duration to drive the timer; everything else (video,
/// description, educational cards) stays phone-only.
///
/// Kept duplicated rather than extracted into a shared module so this
/// wave can ship without restructuring the project; the extraction is
/// a natural follow-up.
struct WatchStretch: Codable, Hashable, Identifiable {
    let id: String
    let name: String
    let durationSeconds: Int
}

struct WatchRoutine: Codable, Hashable {
    let name: String
    let stretches: [WatchStretch]
}
